content/1_docs/2_cookbook/9_setup/0_kirby-in-a-nutshell/recipe.txt

```diff
@@ -0,0 +1,318 @@
+
+- It can affect SEO and usability because of outdated links
+- Internal references are by default stored with their ID. That means, if you change the folder name, the reference you stored will no longer be valid.
+- The same is true if you use the ID to find a page in your templates, blueprints, controllers etc. with a query like `kirby.page("photography")` or its PHP equivalent.
+- It is also not uncommon to use page IDs as class names for styling. If the ID changes, the styles will no longer apply.
+
+There are ways to deal with or prevent such issues, for example:
+
+- To prevent invalid references, you can use custom, unique permanent identifiers that you add manually or you can use a (link: https://github.com/bnomei/kirby3-autoid text: plugin) to create Auto-IDs which are then stored instead of the default ID. This permanent identifier can also be used to query pages.
+- Another option would be to prevent deleting or renaming pages, or by using other references like the intended template etc.
+
+## So many fields
+
+(screenshot: panel-form-fields.png)
+
+### Field names
+Field names can sometimes lead to confusion. Ok, field names may only contain alpha-numeric characters and underscores, that's one thing. But field names can also otherwise be a bit tricky. Let's assume you set up a (link: docs/reference/panel/fields/files text: `files` field) like this in a blueprint:
+
+```yaml
+fields:
+  image:
+    type: files
+    max: 1
+    query: page.images
+```
+
+If you try to call this field in your template like this…
+
+```php
+<?= $page->image()->toFile() ?>
+```
+
+…you will probably be very surprised to get no result at all. That is because the `image()` method is a native Kirby method that will fetch the first image it finds in the file system. And since a file object doesn't have a `toFile()` method, the above code will not echo anything, unless your file meta data has a field with that name. Since there a quite a few native Kirby methods and plugins might add even more, it is easy to accidentally pick such a method as your field name.
+
+So how to prevent such issues? There are basically two ways to deal with this:
+
+- Call such fields via the `content()` method: `$page->content()->image()->toFile()`
+- Prevent such issues by prefixing all your field names: `xy_image` instead of `image`
+
+### Field types, blueprints and templates
+
+Different field types store their content in different ways. While a `text` field stores a string, a `tags` or `multiselect` field stores a comma separated list of values, the `pages/files` fields or a `structure` field store their content in (YAML format, and the (link: text: editor) field stores its content in JSON format. Kirby comes with useful (link: docs/reference/templates/field-methods text: field methods) which convert these different values into something usable in your templates. 
```

Comment:
- There's an open bracket before YAML.
- The editor is not linked.

Reply:
✔︎